### ISSUE TITLE
Runs of pylint on the same codebase produce different results

### ISSUE THREAD

**Issue by simonsmiley** [PyCQA/pylint#1060]
Here are two builds once run using my own gitlab runner and on gitlab's shared runners.
My runner results in passing tests, while the shared runner fails. It doesn't seem to related to the runners. Two devs ran the tests locally and they passed, another got the error.
https://gitlab.com/thelinuxguy/hyperkitty/builds/2814491
https://gitlab.com/thelinuxguy/hyperkitty/builds/2814452

Here is the actual code
https://gitlab.com/thelinuxguy/hyperkitty/blob/master/hyperkitty/models/email.py#L208

and this is how we run our tests
https://gitlab.com/thelinuxguy/hyperkitty/blob/master/tox.ini

Since we are using tox, we always use the latest released version of pylint.

**Comment by sebix** [comment]
I experience the same issue with some SUSE distros, here are the build logs including diffs:
- OpenSUSE Leap: https://build.opensuse.org/package/live_build_log/home:sebix:branches:devel:languages:python3/python3-pylint/openSUSE_Leap_42.1/i586
- SUSE Linux Enterprise 12: https://build.opensuse.org/package/live_build_log/home:sebix:branches:devel:languages:python3/python3-pylint/SLE_12_SP1/x86_64

**Comment by PCManticore** [comment]
Hi, sorry for taking so long to reply.

I will take a look in the following days (currently just triaging some issues).

**Comment by sebix** [comment]
Any updates on this?

**Comment by sebix** [comment]
Here are new results for 1.4.5 and pytest:

openSUSE Leap 42.1: https://build.opensuse.org/package/live_build_log/home:sebix:branches:devel:languages:python3/python3-pylint/openSUSE_Leap_42.1/i586
openSUSE Tumblweed: https://build.opensuse.org/package/live_build_log/home:sebix:branches:devel:languages:python3/python3-pylint/openSUSE_Tumbleweed/i586

**Comment by GliderGeek** [comment]
i am also experiencing differences between results on a gitlab runner and my local pylint run.
any update on this?
@PCManticore: any ways i can help getting closer to a solution?

**Comment by PCManticore** [comment]
Sorry @GliderGeek I didn't have time to check this, nor I'll have in the following weeks. If you can investigate from where this difference comes, that will be hopeful. I don't exactly how to verify that, you might put some debugging statements in the visit methods to see if the order of visiting the nodes affect this somehow (although we should probably have consistent visit ordering).

**Comment by Pierre-Sassoulas** [comment]
Closing as it's probably related to subtle changes in the environment, the interpreter, installed packages, installed packages versions or a combination of those factors. Please provide a dockerfile so we can reproduce easily and open a new issue if you encounter a similar issue.

**Comment by sashokbg** [comment]
Hello, I was preparing to add a message to this thread since I had the same issue and I found out the culprit !

In my CICD the git worktree was completely clean, since it is a fresh checkout. On my local build however when typing "git status" I saw that I had an "out" directory that I forgot to clean at one point. My pylint was going over it which resulted in lower score since lots of repetitions.

So the solution is to configure an exception in your pylintrc if you have local untracked files
